testng/src/main/java/cucumber/api/testng/FeatureResultListener.java

```diff
@@ -1,52 +1,35 @@
 package cucumber.api.testng;
 
 import cucumber.runtime.CucumberException;
-import gherkin.formatter.Reporter;
-import gherkin.formatter.model.Match;
-import gherkin.formatter.model.Result;
+import cucumber.api.Result;
+import cucumber.api.event.EventHandler;
+import cucumber.api.event.EventPublisher;
+import cucumber.api.event.TestStepFinished;
+import cucumber.api.formatter.Formatter;
 
-public class FeatureResultListener implements Reporter {
+public class FeatureResultListener implements Formatter {
     static final String PENDING_STATUS = "pending";
     static final String UNDEFINED_MESSAGE = "There are undefined steps";
     static final String PENDING_MESSAGE = "There are pending steps";
-    private Reporter reporter;
     private boolean strict;
     private Throwable error = null;
+    private final EventHandler<TestStepFinished> testStepFinishedHandler = new EventHandler<TestStepFinished>() {
+        @Override
+        public void receive(TestStepFinished event) {
+            collectError(event.result);
+        }
+    };
 
-    public FeatureResultListener(Reporter reporter, boolean strict) {
-        this.reporter = reporter;
+    public FeatureResultListener(boolean strict) {
         this.strict = strict;
     }
 
     @Override
-    public void after(Match match, Result result) {
-        collectError(result);
-        reporter.after(match, result);
+    public void setEventPublisher(EventPublisher publisher) {
+        publisher.registerHandlerFor(TestStepFinished.class, testStepFinishedHandler);
     }
 
-    @Override
-    public void before(Match match, Result result) {
-        collectError(result);
-        reporter.before(match, result);
-    }
-
-    @Override
-    public void embedding(String mimeType, byte[] data) {
-        reporter.embedding(mimeType, data);
```

Comment:
I can not find where this method has been replaced with an event listener. From what I can see this is not needed as the Runtime takes care of this now and the TestNgReporter doesn't care about embeddings.